src/commands/mod.rs

```diff
@@ -478,16 +482,32 @@ impl DaemonControl {
         // Now compute whether we'll need a change output while automatically selecting coins to be
         // used as input if necessary.
         // We need to get the size of a potential change output to select coins / determine whether
-        // we should include one, so get a change address and create a dummy txo for this purpose.
-        let change_index = db_conn.change_index();
-        let change_desc = self
-            .config
-            .main_descriptor
-            .change_descriptor()
-            .derive(change_index, &self.secp);
+        // we should include one, so get the change address and create a dummy txo for this purpose.
+        // The change address may be externally specified for the purpose of a "sweep": the user
+        // would set the value of some outputs (or none) and fill-in an address to be used for "all
+        // the rest". This is the same logic as for a change output, except it's external.
+        struct InternalChangeInfo {
+            pub desc: descriptors::DerivedSinglePathLianaDesc,
+            pub index: bip32::ChildNumber,
+        }
+        let (change_spk, int_change_info) = if let Some(addr) = change_address {
+            let addr = self.validate_address(addr)?;
+            (addr.script_pubkey(), None)
```

Comment:
Should we check in case one of our own change addresses has been passed in the parameter? This would happen as part of RBF, for example, in order for a replacement transaction to use the same change address.

In the case of RBF, I don't think we would need to increment the next index in the DB, as it should already be bigger than the one we pass. But perhaps we should check that and increment the next index accordingly?

Reply:
Yes, thanks for the reminder i need to address https://github.com/wizardsardine/liana/pull/821#issuecomment-1816362834.

Reply:
Done.